testscript/exe.go

```diff
@@ -49,61 +50,116 @@ func IgnoreMissedCoverage() {
 //
 // This function returns an exit code to pass to os.Exit, after calling m.Run.
 func RunMain(m TestingM, commands map[string]func() int) (exitCode int) {
-	goCoverProfileMerge()
-	cmdName := os.Getenv("TESTSCRIPT_COMMAND")
-	if cmdName == "" {
+	// Depending on os.Args[0], this is either the top-level execution of
+	// the test binary by "go test", or the execution of one of the provided
+	// commands via "foo" or "exec foo".
+
+	rxBinary := regexp.MustCompile(`go-build.*\.test(\.exe)?$`)
+	if rxBinary.MatchString(os.Args[0]) {
+		// Set up all commands in a directory, added in $PATH.
+		// TODO: can we reuse WorkdirRoot somehow?
+		// TODO: can we do better than os.Setenv?
+		bindir, err := ioutil.TempDir("", "testscript-bin")
```

Comment:
I didn't do that because I hoped to just reuse WorkdirRoot. But as far as I know that gets created later. If you confirm it's not possible to reuse that temp dir here, then yeah, I'll set up both under the same extra temp dir.

Reply:
Done the above.